staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -69,6 +69,11 @@ type Reflector struct {
 	lastSyncResourceVersion string
 	// lastSyncResourceVersionMutex guards read/write access to lastSyncResourceVersion
 	lastSyncResourceVersionMutex sync.RWMutex
+
+	// consistentErrorsCh channel on which we send List&Watch related errors
+	// it is meant to represent a gauge, that is a number that can go up and down
+	consistentErrorsCh chan int
```

Comment:
we could also report the actual errors that could be later returned from the health endpoint.

Reply:
I would prefer exporting the data as a metric initially. Why do you want to return this from the health endpoint?

Reply:
In general we would like to return health functions from the individual controllers. This pull helps to detect stuck list/watch calls. Based on that we could kill the faulty controller instead of the whole KCM. Please take a look on https://github.com/kubernetes/kubernetes/pull/71227

Reply:
Interesting, I'll take a closer look. Thanks for the context!